Builds/CMake/RippledSettings.cmake

```diff
@@ -6,6 +6,11 @@ option (assert "Enables asserts, even in release builds" OFF)
 
 option (reporting "Build rippled with reporting mode enabled" OFF)
 
+option (tests "Build tests" ON)
+if (tests)
+  add_compile_definitions(ENABLE_TESTS)
```

Comment:
`add_compile_definitions` is not supported in our minimum supported cmake version. We usually do this with `target_compile_definitions`, like so:
https://github.com/ripple/rippled/blob/7bd5d51e4e4e76a5547051d30b330739618eddb0/Builds/CMake/RippledCore.cmake#L979